### ISSUE TITLE
NotoSansGujarati features dont' build

### ISSUE THREAD

**Issue by rsheeter** [googlefonts/fontc#436]
@simoncozens posted compile times for Noto Sans Gujarati which let me to wonder what fontc would do. Apparently what it woudl do is fail :)

```shell
# what I actually wanted to run
$ hyperfine --warmup 3 --prepare 'rm -rf build/' 'target/release/fontc --source 
$ ~/oss/google_fonts_sources/sources/ofl/notosansgujarati/sources/NotoSansGujarati.glyphs --emit-ir=false'

# to reproduce, starting from root of fontc workspace
$ git clone https://github.com/notofonts/gujarati.git ../notosansgujarati
$ cargo run -p fontc -- --source ../notosansgujarati/sources/NotoSansGujarati.glyphs --emit-ir=false
[2023-09-13T15:34:38.758771Z main fontc::workload ERROR] Be(Features) failed Fea compilation failure Parsing failed with 88 errors
error: ';' Not valid in a lookup block
in  at 426:4
    | 
426 |     ;
    |     ^

...many similar errors...
```

**Comment by rsheeter** [comment]
@cmyr could I impose on you to take a peek?

**Comment by cmyr** [comment]
peeking

**Comment by simoncozens** [comment]
I peeked at this already. :-) The issue is that when *decompiling* binaries to FEA, I often make use of this construction:

```
lookup foo {
    lookupflag 0;
    ; # This is the issue
    ...
} foo;
```

The rationale behind adding the null statement there is that it compiles whether or not there are additional statements (in some cases there are genuinely empty lookups in the binary). To make that more explicit, this compiles:

```
lookup foo {
    lookupflag 0;
    ;
} foo;
```

but this does not:

```
lookup foo {
    lookupflag 0;
} foo;
```

However. This "null statement" only compiles on feaLib's parser; it doesn't parse in makeotf, and reasonably so because it's not mentioned anywhere in the AFDKO spec. This is why I didn't raise a fea-rs bug; because it's arguably a fontTools-only dialect.

**Comment by anthrotype** [comment]
> in some cases there are genuinely empty lookups in the binary

why?

**Comment by simoncozens** [comment]
https://github.com/fonttools/fonttools/pull/642 may be relevant - apparently it used to be OK in makeotf, but clearly at some point they changed it.

**Comment by simoncozens** [comment]
BTW since there is currently no mark features writer in fontc, compile times are not massively relevant yet... :-)

**Comment by rsheeter** [comment]
Since we want to drop fontc in where fontmake is today I tend to think we should support the null statement

**Comment by cmyr** [comment]
yep no stress here... I'll switch to emitting a warning and skipping the empty statements.

**Comment by cmyr** [comment]
fixed by #437
